src/main/java/gregtech/common/covers/filter/FilterTypeRegistry.java

```diff
@@ -42,6 +42,17 @@ public static void registerItemFilter(int id, Class<? extends ItemFilter> itemFi
         itemFilterById.put(id, itemFilterClass);
     }
 
+    public static int getIdForFilter(Object filter) {
+        int id = -1;
+        if (filter instanceof ItemFilter) {
+            id = itemFilterById.inverse().get(filter.getClass());
+        } else if (filter instanceof FluidFilter) {
+            id = fluidFilterById.inverse().get(filter.getClass());
+        }
+        return id;
+    }
+
+    @Deprecated
```

Comment:
The methods with `@Deprecated` annotations here should have an `@ApiStatus.ScheduledForRemoval(inVersion = "some version")` on them, as well as an `@deprecated` javadoc comment stating what to use as a replacement.